core/src/main/java/org/elasticsearch/common/util/concurrent/PrioritizedRunnable.java

```diff
@@ -33,6 +35,15 @@ public static PrioritizedRunnable wrap(Runnable runnable, Priority priority) {
 
     protected PrioritizedRunnable(Priority priority) {
         this.priority = priority;
+        creationDate = System.nanoTime();
+    }
+
+    public long getCreationDateInNanos() {
+        return creationDate;
+    }
+
+    public long getAgeInNanos() {
+        return Math.max(0, (System.nanoTime() - creationDate));
```

Comment:
can we make this getAgeInMillis and do the conversions here? we make use of millis everywhere so it will make for an easier to work with api. The nanos resolution is not guaranteed anyway.